backend/src/main/java/com/aljoschazoeller/backend/person/PersonService.java

```diff
@@ -0,0 +1,30 @@
+package com.aljoschazoeller.backend.person;
+
+import com.aljoschazoeller.backend.exceptions.PersonNotFoundException;
+import com.aljoschazoeller.backend.person.domain.Person;
+import com.aljoschazoeller.backend.person.domain.PersonStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PersonService {
+    private final PersonRepository personRepository;
+
+    public PersonService(PersonRepository personRepository) {
+        this.personRepository = personRepository;
+    }
+
+    public List<Person> getAllActivePeople() {
+        return personRepository.findAllByStatus(PersonStatus.ACTIVE);
+    }
+
+    public Person getPersonById(String id) {
+        return personRepository.findById(id)
+                .orElseThrow(() -> new PersonNotFoundException("No person found with ID '" + id + "'."));
+    }
+
+    public Person createPerson(Person person) {
+        return personRepository.save(person);
```

Comment:
Wieso kein insert?

Reply:
Danke für den Tipp. Insert wirft tatsächlich eine Exception, wenn (aus versehen) eine ID mitgegeben wird und in der DB ein Element mit dieser ID existiert. Ist hier passender. Sollte ich in der Methode nochmal vorher prüfen ob die ID wirklich NULL ist (ich übergebe mein DTO in die Domainklasse in der Regel im Controller) oder ist das overengineered?

Reply:
Gute Idee. Dann kommt allerdings leider diese SonarLint Meldung:
> Asserts should not be used to check the parameters of a public method
> 
> An assert is inappropriate for parameter validation because assertions can be disabled at runtime in the JVM, meaning that a bad operational setting would completely eliminate the intended checks. Further, asserts that fail throw AssertionErrors, rather than throwing some type of Exception. Throwing Errors is completely outside of the normal realm of expected catch/throw behavior in normal programs.
This rule raises an issue when a public method uses one or more of its parameters with asserts.

Reply:
Mhm komisch. Ich habe auch ein assert bei mir im Code und da kam diese Fehlermeldung nicht (auch eine public methode):

```java
public class UserService {
    public TicketScoutUser getCurrentUser() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        assert authentication.getPrincipal() instanceof OAuth2User;
        OAuth2User oauthUser = (OAuth2User) authentication.getPrincipal();
        return new TicketScoutUser(
                oauthUser.getAttributes().get("login").toString(),
                oauthUser.getAttributes().get("avatar_url").toString()
        );
    }
```

Reply:
Aber hier hast Du ja garkeine Argumente, die entgegengenommen werden und die Du im assert prüfst. Von daher keine Abhängigkeit von außen.

Reply:
In Java, the use of assertions to check the parameters of a public method is considered bad practice because assertions can be disabled at runtime, leading to inconsistent behavior. Instead, you should use explicit checks and throw appropriate exceptions to ensure that invalid parameters are handled correctly. Here's how you can fix this issue:

1. Use Exceptions for Parameter Validation: Replace assertions with explicit parameter validation using exceptions like IllegalArgumentException, NullPointerException, or custom exceptions.

### Example:
Suppose you have a method that looks like this, using assertions:

```java
public void setAge(int age) {
    assert age > 0 : "Age must be positive";
    this.age = age;
}
```

You should replace the assertion with an explicit check:

```java
public void setAge(int age) {
    if (age <= 0) {
        throw new IllegalArgumentException("Age must be positive");
    }
    this.age = age;
}
```

Reply:
Ja richtig, das ist die Fehlermeldung. Aber sollte ich in meinem Fall wirklich prüfen, ob die ID NULL ist und dann eine Exception werfen?

Reply:
Hab ich in https://github.com/josch87/Cinetiq/pull/32/commits/95cd55c96748d328fe1ee5eb20c7f852b77c0d83 jetzt einfach mal umgesetzt.